webapp/app/components/pages/ActionButtons/ActionButtons.jsx

```diff
@@ -2,161 +2,215 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { Button } from 'reactstrap';
 import { connect } from 'react-redux';
-import { cloneDeep, isEmpty } from 'lodash/fp';
-import {bindActionCreators} from 'redux';
+import isEmpty from 'lodash/isEmpty';
+import has from 'lodash/has';
 import MsModal from '../../common/modal/MsModal';
 import UsersForm from '../UsersForm/UsersForm';
-import * as usersActions from '../../../actions/usersActions';
-import { EMAIL_REGEX } from '../../../constants';
+import { EMAIL_REGEXP } from '../../../constants';
 
 const EMPTY_USER = {
-  id: '',
   name: '',
   email: '',
   phone: '',
   skypeId: ''
 };
 
-const FORM_ERRORS = {
-  name: '',
-  email: {
-    invalid: '',
-    missing: ''
-  }
+const DEFAULT_USER_MODAL_LABELS = {
+  confirmButtonText: 'Save'
 };
 
+function isUserMatchById(sourceUser = {}, targetUser = {}) {
+  return sourceUser['id'] === targetUser['id']; 
+}
+
+function isValidUser(user) {
+  return has(user, 'id') && isEmpty(user, 'id') === false;
+}
+
 export class ActionButtons extends React.Component {
   constructor(props) {
     super(props);
     
     this.state = {
+      actionType: null,
       user: {...EMPTY_USER, ...props.user},
-      modal: false,
+      isUserModalOpen: false,
       modalTitle: '',
       modalBody: {},
       modalYesLabel: '',
-      errors: cloneDeep(FORM_ERRORS)
+      errors: {}
     };
 
     this.toggle = this.toggle.bind(this);
     this.toggleAddModal = this.toggleAddModal.bind(this);
     this.toggleEditModal = this.toggleEditModal.bind(this);
     this.toggleDeleteModal = this.toggleDeleteModal.bind(this);
-    this.noUserSelected = this.noUserSelected.bind(this);
     this.updateUserState = this.updateUserState.bind(this);
     this.saveUser = this.saveUser.bind(this);
     this.cancel = this.cancel.bind(this);
-    this.isFormValid = this.isFormValid.bind(this);
+    this.validateForm = this.validateForm.bind(this);
+    this.renderUserModalBody = this.renderUserModalBody.bind(this);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.props.user.id !== nextProps.user.id) {
+  componentWillReceiveProps({ user: newUser }) {
+    const { user: currentSelectedUser } = this.props;
+
+    if (!isUserMatchById(currentSelectedUser, newUser)) {
       this.setState({
         user: {
-          ...nextProps.user
+          ...newUser
         }
       });
     }
   }
 
   toggle() {
     this.setState({
-      modal: !this.state.modal
+      isUserModalOpen: !this.state.isUserModalOpen
     });
   }
 
   toggleAddModal() {
     this.setState({
-      user: EMPTY_USER,
-      modalBody:'',
-      modalTitle: 'Add User',
-      modalYesLabel: 'Save'
+      actionType: 'add',
+      user: EMPTY_USER
```

Comment:
You must clone empty user.

Reply:
correct, thanks it should be `{ ...EMPTY_USER }`